codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -852,23 +864,92 @@ void generateReferencedResources(GenerationContext context) {
-                    """
-                    var native_response = %s(dafny_response.Dtor_value().(%s))
-                    return &native_response, nil
-                    """.formatted(
-                        SmithyNameResolver.getFromDafnyMethodName(
+                  if (outputShape.hasTrait(PositionalTrait.class)) {
+                    final MemberShape postionalMemShape = outputShape
+                      .getAllMembers()
+                      .values()
+                      .stream()
+                      .findFirst()
+                      .get();
+                    outputShape =
+                      model.expectShape(postionalMemShape.getTarget());
+                    if (outputShape.hasTrait(ReferenceTrait.class)) {
+                      outputShape =
+                        model.expectShape(
+                          outputShape
+                            .expectTrait(ReferenceTrait.class)
+                            .getReferentId()
+                        );
+                    }
+                    String fromDafnyConvMethodName =
+                      outputShape.isResourceShape()
+                        ? SmithyNameResolver.getFromDafnyMethodName(
                           service,
                           outputShape,
                           ""
-                        ),
-                        DafnyNameResolver.getDafnyType(
-                          inputShape,
+                        )
+                        : Constants.funcNameGenerator(
+                          postionalMemShape,
+                          "FromDafny",
+                          model
+                        );
+
+                    outputType =
+                      SmithyNameResolver
+                        .getSmithyType(
+                          outputShape,
                           symbolProvider.toSymbol(outputShape)
                         )
-                      );
-                  returnError = "return nil,";
+                        .concat(",");
+                    String deReferenceRequired = "";
+                    if (outputShape.hasTrait(ServiceTrait.class)) {
+                      deReferenceRequired = "*";
+                    }
+                    var typeAssertion = outputShape.isResourceShape()
+                      ? ".(%s)".formatted(
+                          DafnyNameResolver.getDafnyType(
+                            outputShape,
+                            symbolProvider.toSymbol(outputShape)
+                          )
+                        )
+                      : "";
+                    returnResponse =
+                      """
+                      var native_response = %s(dafny_response.Dtor_value()%s)
+                      return %snative_response, nil
+                      """.formatted(
+                          fromDafnyConvMethodName,
+                          typeAssertion,
+                          deReferenceRequired
```

Comment:
Do we always return a value type for Positional Output? Otherwise this might fail.

Reply:
https://github.com/smithy-lang/smithy-dafny/issues/727